tests/testinterop.nim

```diff
@@ -105,18 +106,19 @@ proc testPubSubDaemonPublish(gossip: bool = false,
   let awaiters = nativeNode.start()
   let nativePeer = nativeNode.peerInfo
 
-  var finished = false
+  var finished = newFuture[void]()
   var times = 0
   proc nativeHandler(topic: string, data: seq[byte]) {.async.} =
     let smsg = cast[string](data)
     check smsg == pubsubData
     times.inc()
-    if times >= count and not finished:
-      finished = true
+    if times >= count and not finished.finished:
```

Comment:
This does't seem like it makes any difference, `finished` is sync and has the same effect as the bool flag, same with the future, it is only used after `published` exits when it's already resolved, so this has the same effect but with the overhead of an additional future.